src/main/java/seedu/address/model/animal/Animal.java

```diff
@@ -104,6 +132,13 @@ public String getBreedForSerialization() {
         return breed.toString();
     }
 
+    public List<JsonAdaptedTask> getTaskListForSerialization() {
+        return taskList.getTaskList()
+                .stream()
+                .map(JsonAdaptedTask::new)
+                .collect(Collectors.toList());
+    }
+
```

Comment:
Nit but this method's logic should be moved to TaskList

Reply:
Oki